src/pages/registration/new_registration/LinkRegistration.tsx

```diff
@@ -1,67 +1,115 @@
+import { RegistrationList } from '../../../components/RegistrationList';
 import { setNotification } from '../../../redux/notificationSlice';
-import { Doi } from '../../../types/registration.types';
-import { isErrorStatus, isSuccessStatus } from '../../../utils/constants';
 import { dataTestId } from '../../../utils/dataTestIds';
+import { isValidUrl } from '../../../utils/general-helpers';
 import { stringIncludesMathJax, typesetMathJax } from '../../../utils/mathJaxHelpers';
 import { getRegistrationWizardPath } from '../../../utils/urlPaths';
-import { LinkRegistrationForm } from './LinkRegistrationForm';
 import { RegistrationAccordion } from './RegistrationAccordion';
 
 export interface StartRegistrationAccordionProps {
   expanded: boolean;
   onChange: (event: ChangeEvent<unknown>, isExpanded: boolean) => void;
 }
 
+enum LinkRegistrationFormFieldName {
+  Link = 'link',
+}
+
+const doiValidationSchema = Yup.object({
+  [LinkRegistrationFormFieldName.Link]: Yup.string().trim().required(),
+});
+
+interface DoiFormValues {
+  [LinkRegistrationFormFieldName.Link]: string;
+}
+
+const emptyDoiFormValues: DoiFormValues = {
+  [LinkRegistrationFormFieldName.Link]: '',
+};
+
+const doiUrlBase = 'https://doi.org/';
+const doiUrlPlaceholder = `${doiUrlBase}10.1000/xyz123`;
+const doiRegExp = new RegExp('\\b(10[.][0-9]{4,}(?:[.][0-9]+)*/(?:(?!["&\'<>])\\S)+)\\b'); // https://stackoverflow.com/a/10324802
+
 export const LinkRegistration = ({ expanded, onChange }: StartRegistrationAccordionProps) => {
   const { t } = useTranslation();
-  const [doi, setDoi] = useState<Doi | null>(null);
-  const [noHit, setNoHit] = useState(false);
+  const [doiQuery, setDoiQuery] = useState('');
   const history = useHistory();
   const dispatch = useDispatch();
 
   const openRegistration = async () => {
-    if (!doi) {
+    if (!mutateDoi.data) {
       return;
     }
-    history.push(getRegistrationWizardPath(doi.identifier), { highestValidatedTab: -1 });
+    history.push(getRegistrationWizardPath(mutateDoi.data.data.identifier), { highestValidatedTab: -1 });
   };
 
-  const handleSearch = async (doiUrl: string) => {
-    setNoHit(false);
-    setDoi(null);
-
-    try {
-      const doiRegistrationResponse = await getRegistrationByDoi(doiUrl);
-      if (isErrorStatus(doiRegistrationResponse.status)) {
-        setNoHit(true);
-        dispatch(setNotification({ message: t('feedback.error.get_doi'), variant: 'error' }));
-      } else if (isSuccessStatus(doiRegistrationResponse.status)) {
-        if (doiRegistrationResponse.data) {
-          setDoi(doiRegistrationResponse.data);
-        } else {
-          setNoHit(true);
+  const resultsQuery = useQuery({
+    enabled: !!doiQuery,
+    queryKey: ['doi-results', doiQuery],
+    queryFn: () => fetchResults2(10, 0, `"${doiQuery}"`),
+    onSuccess: (response) => {
+      if (response.hits.length === 0) {
+        mutateDoi.mutate(doiQuery);
+      }
+    },
```

Comment:
brukte onSuccess her bare for å få noe som fungerer, så kan se om vi får flyttet dette til useEffect også da, for å ikke lage nye ting som allerede er deprecated

Reply:
Skal funke fint med useEffect nå